pkg/acsengine/defaults.go

```diff
@@ -296,6 +306,10 @@ func setAgentNetworkDefaults(a *api.Properties) {
 			profile.OSType = api.Linux
 		}
 
+		if a.OrchestratorProfile.OrchestratorType == api.Kubernetes && a.HostedMasterProfile != nil {
```

Comment:
I am assuming the RP sets the master subnet value somehow during conversion from 20170831 to pkg/api/types.go and then agents get these values? I wouldn't recommend doing it this way.

Another way to do this is have the default value in: github.com\Azure\acs-engine\pkg\acsengine\const.go and then set that value for both agent (if VnetSubnetID is empty) and master subnet.